### ISSUE TITLE
False positives in skip check (after merge on top of skipped commit)

### ISSUE THREAD

**Issue by h-vetinari** [conda-forge/conda-smithy#2139]
I've run into the following kind of situation a few times now, where a merge commit gets skipped if the last commit on main contained `[ci skip]` etc. The following is a shortened & annotated `git log --decorate --oneline --graph --all` from https://github.com/conda-forge/ctng-compiler-activation-feedstock/, where

```
*   ddc2d49 (up/main) Merge pull request #136 from minrk/isystem-not-enough     # CI ran
|\
| * 4e19877 add -I $PREFIX/include to FFLAGS
* |   3648ba7 Merge pull request #131 from h-vetinari/clang19                   # CI skipped!
|\ \
| * | 68e8776 (origin/clang19, clang19) readd libstdcxx-devel to clangxx_impl; fix typo
| * | 111e074 [...]
| |/
* / 777282f [ci skip] [skip ci] [cf admin skip] ***NO_CI*** admin migration RemoveAutomergeAndRerender
|/
*   9833f59 Merge pull request #135 from apmorton/am/gcc-14.2                   # CI ran
```

In particular, the merge after https://github.com/conda-forge/ctng-compiler-activation-feedstock/pull/131 seems to falsely determine that the [job](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1081538&view=results) should be skipped. Presumably there's an issue with how skips get identified for merge commits, in that it wrongly picks up on the last non-merge commit on main (or something like that)?

CC @beckermr @jaimergp

**Comment by jaimergp** [comment]
I think @beckermr had removed the skip checks in `main` 🤔 See https://github.com/conda-forge/conda-smithy/pull/2110. It looks like that [file has not been rerendered in a while](https://github.com/conda-forge/ctng-compiler-activation-feedstock/blob/ddc2d498b0a04987a2fc5f315082a52a74a23000/azure-pipelines.yml#L22), though.

**Comment by beckermr** [comment]
Yes please rerender and try again.

**Comment by h-vetinari** [comment]
That feedstock currently cannot be rerendered due to https://github.com/conda-forge/conda-smithy/issues/2100. I did see it elsewhere as well, but will definitely check if a rerender solves it when I next encounter it.

**Comment by h-vetinari** [comment]
Another example that wrongly gets skipped (though this case is arguably hard to avoid, because `[ci skip]` could presumably _legitimately_ not be at the beginning): `Revert "[ci skip] disable optional test dep sparse to unblock 3.13 migration`, see [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1089526&view=results)

**Comment by beckermr** [comment]
Yeah and "Revert" could legitimately be part of a commit one would want to skip Ci for too. There is nothing we can do about this one unless we can get GitHub to read our minds.
